airflow/ui/src/pages/Dashboard/Metrics.tsx

```diff
@@ -0,0 +1,244 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  Box,
+  Flex,
+  Heading,
+  HStack,
+  VStack,
+  Text,
+  createListCollection,
+  type SelectValueChangeDetails,
+} from "@chakra-ui/react";
+import dayjs from "dayjs";
+import type {
+  TaskInstanceStateCount,
+  DAGRunStates,
+} from "openapi-gen/requests/types.gen";
+import { useState } from "react";
+import { FiCalendar } from "react-icons/fi";
+
+import { useDashboardServiceHistoricalMetrics } from "openapi/queries";
+import { ErrorAlert } from "src/components/ErrorAlert";
+import Time from "src/components/Time";
+import { Select } from "src/components/ui";
+import { capitalize } from "src/utils";
+import { stateColor } from "src/utils/stateColor";
+
+const BAR_WIDTH = 100;
+const BAR_HEIGHT = 5;
+const DAGRUN_STATES: Array<keyof DAGRunStates> = [
+  "queued",
+  "running",
+  "success",
+  "failed",
+];
+const TASK_STATES: Array<keyof TaskInstanceStateCount> = [
+  "queued",
+  "running",
+  "success",
+  "failed",
+  "skipped",
+];
+const timeOptions = createListCollection({
+  items: [
+    { label: "Last 1 hour", value: "1" },
+    { label: "Last 8 hours", value: "8" },
+    { label: "Last 12 hours", value: "12" },
+    { label: "Last 24 hours", value: "24" },
+  ],
+});
+
+type Props = {
+  readonly runs: number;
+  readonly state: string;
+  readonly total: number;
+};
+
+type DagRunStateInfoProps = {
+  readonly dagRunStates: DAGRunStates;
+  readonly total: number;
+};
+
+type TaskRunStateInfoProps = {
+  readonly taskRunStates: TaskInstanceStateCount;
+  readonly total: number;
+};
+
+const StateInfo = ({ runs, state, total }: Props) => {
```

Comment:
We should make a `/metrics` sub directory and make each component its own file. Then hot-refresh can work most efficiently and we split up our code.